server/src/main/java/org/candlepin/model/OwnerProductShareCurator.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2009 - 2017 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.model;
+
+import com.google.inject.Inject;
+import com.google.inject.persist.Transactional;
+
+import org.apache.commons.collections.CollectionUtils;
+
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+
+import javax.persistence.TypedQuery;
+
+/**
+ * Curator to handle creation and maintenance of OwnerProductShare objects
+ */
+public class OwnerProductShareCurator extends AbstractHibernateCurator<OwnerProductShare> {
+
+    @Inject
+    public OwnerProductShareCurator() {
+        super(OwnerProductShare.class);
+    }
+
+    public List<OwnerProductShare> findProductSharesByRecipient(Owner owner, boolean activeOnly,
+        Collection<String> productIds) {
+        return findProductSharesByOwner(owner, true, activeOnly, productIds);
+    }
+
+    public List<OwnerProductShare> findProductSharesBySharer(Owner owner, boolean activeOnly,
+        Collection<String> productIds) {
+        return findProductSharesByOwner(owner, false, activeOnly, productIds);
+    }
+
+    private List<OwnerProductShare> findProductSharesByOwner(Owner owner, boolean isRecipient,
```

Comment:
I don't think it's a great idea to have the meaning of the method call hinge on a boolean like this.  Why not have a `findProductSharesGivenToOwner` and a `findProductSharesMadeByOwner` to encapsulate the difference between

```java
if (isRecipient) {
    jpql += "ps.recipientOwner.id = :owner_id ";
}
else {
    jpql += "ps.sharingOwner.id = :owner_id ";
}
```

Otherwise, it becomes a constant guessing game when a developer is reading the code as to what exactly they're getting back.

Reply:
note from discussion, this is private method, so its okay.